### PR TITLE
[WIP] Include propsToList R function for dash-docs

### PR DESCRIPTION
**Pull request by rpkyle**
This PR proposes to update `utils.R` with a function entitled `propsToList`. 

This function will generate a list of elements containing metadata for component properties, simplifying and systematizing the generation of tables for online Dash component documentation. This metadata is abstracted from the help files within R packages, e.g. `dccGraph.Rd`, as seen in its internal help viewer.

Currently, `attribute`, `description` and `type` are supported; support for extracting `default` values is forthcoming.

Related to https://github.com/plotly/dash-docs/pull/479. Closes #462.

@chriddyp @alexcjohnson @KPhans

### CODE REVIEW

**Comment by rpkyle** [comment]
@alexcjohnson This PR has gone a bit stale, but is still relevant and the code is working. 

Any chance you'd be able to review and help me merge this thing? 🙏

**Comment by rpkyle** [comment]
Closes #469.

**Comment by rpkyle** [comment]
@alexcjohnson If this looks 👌, I'll work on merging it in to `dashr-squashed` so we can close #469 too.

File: dashr/utils.R [hunk 1]
```diff
@@ -89,20 +89,27 @@ LoadAndDisplayComponent2 <- function(example_string) {
...
+  list_of_args <- lapply(unlist(args, 
+                                recursive=FALSE), 
                          function(x) x[x != "\n"])
-  # Extract the prop metadata for tabulation
+
+  # Extract the prop metadata for tabulation  
+                         
   props_as_list <- invisible(sapply(list_of_args, function(x) {
     if (any(sapply(x, is.list))) {
       all_types <- paste("List of numerics",
```
Comment (HammadTheOne) [inline dashr/utils.R]: May want to add `List of lists` as a generic option as well to the types. When testing with `dccChecklist`, the `helpfile` describes the `options` argument as `List of lists`, but the `type` key in the resulting list simply has a value of `list`.

**Pull request closed without merging**